src/client/cache.rs

```diff
@@ -54,12 +54,14 @@ impl ConfigurationSnapshot {
         // FIXME: why not filtering for collection here?
 
         let mut features = HashMap::new();
-        for feature in environment.features {
+        for mut feature in environment.features {
+            feature.segment_rules.sort_by(|a, b| a.order.cmp(&b.order));
             features.insert(feature.feature_id.clone(), feature);
         }
 
         let mut properties = HashMap::new();
-        for property in environment.properties {
+        for mut property in environment.properties {
+            property.segment_rules.sort_by(|a, b| a.order.cmp(&b.order));
```

Comment:
One drawback is, that with this all rules sent by server are sorted., while previously the sorting happened on demand only related to features requested by the user.

But I guess the overhead should be quite minimal. It all depends on the assumption of how this is used.

Reply:
I agree. It is a balance between the number of features and the number of times the same feature is evaluated. IMHO, each feature will be evaluated many more times (the murmur hash algorithm is based on that premise) than the number of features in the configuration.